File: examples/python/1-simple-chat-via-apim.py
"""
Example 1: Simple Chat via APIM Gateway
=========================================

This is the simplest possible example. Developers just need:
1. Azure AI project ID (provided by IT)
2. APIM gateway URL (provided by IT)
3. Their corporate identity (they already have it!)

No API keys to manage. No direct Azure OpenAI endpoints. Just simple, governed AI.
"""

from azure.ai.projects import AIProjectClient
from azure.identity import DefaultAzureCredential

# Configuration (your IT manager provides these)
PROJECT_ID = "my-hub-project"  # Ask IT for your project ID
APIM_GATEWAY_URL = "https://your-company-ai.azure-api.net"

def simple_chat():
    """Ask a simple question."""
    
    # 1. Authenticate using corporate identity
    credential = DefaultAzureCredential()
    
    # 2. Create client pointing to your APIM gateway (NOT direct Azure OpenAI)
    client = AIProjectClient(
        credential=credential,
        project_id=PROJECT_ID,
        endpoint=APIM_GATEWAY_URL
    )
    
    # 3. Get the inference client
    chat_client = client.inference.get_chat_completions_client()
    
    # 4. Send a message
    response = chat_client.complete(
        model="gpt-4o",  # Any model available in your Foundry project
        messages=[
            {"role": "user", "content": "What is the capital of France?"}
        ]
    )
    
    # 5. Print the response
    print(f"Response: {response.choices[0].message.content}")
    print(f"Tokens used: {response.usage.total_tokens}")

if __name__ == "__main__":
    simple_chat()

File: examples/csharp/1-simple-chat-via-apim.cs
using Azure.AI.Projects;
using Azure.AI.Inference;
using Azure.Identity;
using Microsoft.Agents.AI;

/// <summary>
/// Example 1: Simple Chat via APIM Gateway (.NET)
/// 
/// For .NET developers using the Agent Framework.
/// Just point to your APIM gateway URL instead of direct Azure OpenAI.
/// </summary>

class SimpleChatExample
{
    private const string ProjectId = "my-hub-project";
    private const string ApimGatewayUrl = "https://your-company-ai.azure-api.net";

    static async Task Main(string[] args)
    {
        await RunSimpleChatAsync();
    }

    static async Task RunSimpleChatAsync()
    {
        // 1. Use DefaultAzureCredential (your corporate identity)
        var credential = new DefaultAzureCredential();

        // 2. Point to APIM gateway URL, NOT direct Azure OpenAI
        var client = new AIProjectClient(
            new Uri(ApimGatewayUrl),
            credential
        );

        // 3. Get the chat client
        var chatClient = client.GetChatClient("gpt-4o");

        // 4. Send a message
        var response = await chatClient.CompleteAsync(
            messages: new[]
            {
                new ChatMessage { Role = ChatCompletionRole.User, Content = "What's the capital of France?" }
            }
        );

        // 5. Print response
        Console.WriteLine($"Response: {response.Value.Content[0].Text}");
        Console.WriteLine($"Tokens: {response.Value.Usage.TotalTokens}");
    }
}
